go/vt/vtgate/safe_session.go

```diff
@@ -597,10 +597,10 @@ func (session *SafeSession) SetPreQueries() []string {
 	first := true
 	for _, k := range keys {
 		if first {
-			preQuery.WriteString(fmt.Sprintf("set @@%s = %s", k, sysVars[k]))
+			preQuery.WriteString(fmt.Sprintf("set %s = %s", k, sysVars[k]))
```

Comment:
Why is this change needed?

Reply:
TLDR; previously it was broken. 

@@ without any scope represents `session` scope. Only for the case of `transaction_isolation` it means the next transaction. This gets applied only once. 
With the implementation of `settings pool` these settings will not be applied again and the connection will be reused,  which means the next query the connection will see, will have the session level isolation level and not the next transaction isolation level.
So, the user expectation will turn out to be wrong.

The RFC already states that the next transaction isolation statements will be converted to the session level setting.